base/setup/lib/setuplib.c

```diff
@@ -649,23 +655,50 @@ InitDestinationPaths(
 /** Equivalent of 'NTOS_INSTALLATION::SystemArcPath' **/
     /* Create 'pSetupData->DestinationArcPath' */
     RtlFreeUnicodeString(&pSetupData->DestinationArcPath);
-    RtlStringCchPrintfW(PathBuffer, ARRAYSIZE(PathBuffer),
-            L"multi(0)disk(0)rdisk(%lu)partition(%lu)\\",
-            DiskEntry->BiosDiskNumber,
-            PartEntry->OnDiskPartitionNumber);
-    ConcatPaths(PathBuffer, ARRAYSIZE(PathBuffer), 1, InstallationDir);
-    RtlCreateUnicodeString(&pSetupData->DestinationArcPath, PathBuffer);
+    Status = RtlStringCchPrintfW(PathBuffer, ARRAYSIZE(PathBuffer),
+                     L"multi(0)disk(0)rdisk(%lu)partition(%lu)\\",
+                     DiskEntry->BiosDiskNumber,
+                     PartEntry->OnDiskPartitionNumber);
+
+    /* Check the condition status of RtlStringCchPrintfW() */
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT("RtlStringCchPrintfW() failed with status 0x%08lx\n", Status);
```

Comment:
The RtlCreateUnicodeString call above allocates memory, so you have one leak here and at least one in every error exit case below.